dockerfiles/git.Dockerfile

```diff
@@ -0,0 +1,24 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.18
+ARG ALPINE_VERSION=3.16
+ARG GIT_VALIDATION_VERSION=v1.1.0
+
+FROM alpine:${ALPINE_VERSION} AS base
+RUN apk add --no-cache git gpg
+
+FROM golang:${GO_VERSION}-alpine AS gitvalidation
+ARG GIT_VALIDATION_VERSION
+RUN --mount=type=cache,target=/root/.cache \
+  --mount=type=cache,target=/go/pkg/mod \
+  GOBIN=/out go install "github.com/vbatts/git-validation@${GIT_VALIDATION_VERSION}"
```

Comment:
nit-picky, but perhaps you could align the flags; it sometimes helps to see differences, and makes the formatting slightly less cluttered), e.g.

```dockerfile
RUN --mount=type=cache,target=/root/.cache \
    --mount=type=cache,target=/go/pkg/mod \
      GOBIN=/out go install "github.com/vbatts/git-validation@${GIT_VALIDATION_VERSION}"
```